x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/parser/EqlParser.java

```diff
@@ -56,11 +57,11 @@ public Expression createExpression(String expression) {
         return invokeParser(expression, EqlBaseParser::singleExpression, AstBuilder::expression);
     }
 
-    private <T> T invokeParser(String sql,
+    private <T> T invokeParser(String eql,
             Function<EqlBaseParser, ParserRuleContext> parseFunction,
                                BiFunction<AstBuilder, ParserRuleContext, T> visitor) {
         try {
-            EqlBaseLexer lexer = new EqlBaseLexer(new CaseInsensitiveStream(sql));
+            EqlBaseLexer lexer = new EqlBaseLexer(new ANTLRInputStream(eql));
```

Comment:
Without the case insensitive stream the query `foo WHeRE` would be invalid. In fact with the current grammar, any keyword not lowercase would fail.
Note that the stream is case insensitive for symbols and it doesn't interfere with the identifiers, qualified or not.

Reply:
This was intentional. All of the keywords in EQL have historically been lowercased, and I don't see any problems that this would impose. It also helps with consistent style by having less options. Otherwise some users write `WHERE` and others write `where` and some writing `Where`.

If we all of the sudden add case-insensitivity to tokens, then rules written in the form `process WHERE x` will only be compatible with the Elasticsearch implementation of EQL, and not the [endgameinc/eql](https://github.com/endgameinc/eql) version of EQL or the one within the Endgame platform.

I think we should have our implementations as consistent as possible, with exceptions only when necessary.